apps/checkout-app/frontend/components/elements/AppProvider/AppProvider.tsx

```diff
@@ -1,19 +1,39 @@
+import { isSsr } from "@/constants";
 import { createApp } from "@saleor/app-bridge";
-import { createContext, useMemo } from "react";
+import { createContext, useEffect, useMemo, useState } from "react";
 
 interface IAppContext {
   app?: any;
+  token?: string;
 }
 
-export const AppContext = createContext<IAppContext>({ app: undefined });
+export const AppContext = createContext<IAppContext>({
+  app: undefined,
+  token: undefined,
+});
 
 const AppProvider: React.FC = (props) => {
+  const [token, setToken] = useState<string>();
```

Comment:
Token should not be kept in React state, instead you can call `createApp` outside the component and use `app.getState().token` inside client